vyper/parser/parser_utils.py

```diff
@@ -263,6 +263,18 @@ def getpos(node):
     )
 
 
+def set_offsets(node, pos):
+    for field in node.get_slots():
+        item = getattr(node, field, None)
+        if isinstance(item, ast.VyperNode):
+            set_offsets(item, pos)
+        elif isinstance(item, list):
+            for i in item:
+                if isinstance(i, ast.VyperNode):
+                    set_offsets(i, pos)
+    node.lineno, node.col_offset, node.end_lineno, node.end_col_offset = pos
```

Comment:
@fubuloubu @iamdefinitelyahuman Adding a NodeVisitor to for VyperNodes would be a separate PR IMHO.
It's a good idea to have a `from vyper.ast import VyNodeVisitor` class (that would be almost a copy of the py ast one) in our codebase - mainly because it allows us to customise. And the python implementation is hidden except for the docs showing usage.